app-modules/task/src/Imports/TaskImporter.php

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * @property ?Task $record
+ */
+#[AllowDynamicProperties]
+class TaskImporter extends Importer
+{
+    /**
+     * @inheritDoc
+     */
+    public static function getColumns(): array
+    {
+        return [
+            ImportColumn::make('title')
+                ->requiredMapping()
+                ->example('The task title'),
+            ImportColumn::make('description')
+                ->requiredMapping()
+                ->example('A description of the task.'),
+            ImportColumn::make('status')
+                ->requiredMapping()
+                ->rules([new Enum(TaskStatus::class)])
+                ->example('pending'),
+            ImportColumn::make('due')
+                ->rules(['date_format:Y-m-d H:i:s'])
+                ->example(fake()->dateTimeBetween('+1 day', '+1 week')->format('Y-m-d H:i:s')),
+            ImportColumn::make('assignedTo')
+                ->relationship(
+                    resolveUsing: fn (mixed $state) => User::query()
+                        ->when(
+                            Str::isUuid($state),
+                            fn (Builder $query) => $query->whereKey($state),
+                            fn (Builder $query) => $query->where('email', $state),
+                        )
+                        ->first(),
+                )
+                ->requiredMapping()
+                ->example(fn (): ?string => auth()->user()?->email ?? User::query()->value('email')),
+            ImportColumn::make('concern')
+                ->relationship(
+                    resolveUsing: function (mixed $state) {
+                        $type = str($state)->before(':');
+                        $value = str($state)->after(':');
+
+                        return match ($type->toString()) {
+                            'prospect' => Prospect::query()
+                                ->when(
+                                    str($value)->isUuid(),
+                                    fn (Builder $query) => $query->whereKey($value),
+                                    fn (Builder $query) => $query->where('email', $value),
+                                )
+                                ->first(),
+                            'student' => Student::query()
+                                ->when(
+                                    str($value)->isUuid(),
+                                    fn (Builder $query) => $query->whereKey($value),
+                                    fn (Builder $query) => $query->where('email', $value),
+                                )
+                                ->first(),
+                        };
+                    },
+                )
+                ->requiredMapping()
+                ->rules(
+                    [
+                        'starts_with:prospect:,student:',
+                    ]
+                )
+                ->example(fn (): string => 'student:' . Student::query()->value('email') ?? fake()->safeEmail()),
```

Comment:
I believe this will fail in production because `composer install --no-dev` will not install faker.